src/TLuaInterpreter.cpp

```diff
@@ -3562,70 +3594,116 @@ int TLuaInterpreter::createMiniConsole(lua_State* L)
+                // window case - as is the case for an empty string
+                luaSendWindow.clear();
+            }
     }
-    int x, y, width, height;
-    bool fillBackground, clickthrough = false;
+
     if (!lua_isnumber(L, 2)) {
-        lua_pushstring(L, "createLabel: wrong argument type");
-        lua_error(L);
-        return 1;
-    } else {
-        x = lua_tonumber(L, 2);
+        luaSendText = lua_tostring(L, 2);
+        if(!lua_isstring(L,2)){
+            lua_pushstring(L, "createLabel: wrong argument type");
+            lua_error(L);
+            return 1;}
+     }else {
+        luaSendText = luaSendWindow;
+        luaSendWindow.clear();
+        counter = 2;
     }
-    if (!lua_isnumber(L, 3)) {
+
+    bool fillBackground, clickthrough = false;
+       if (!lua_isnumber(L, counter)) {
+            lua_pushstring(L, "createLabel: wrong argument type");
+            lua_error(L);
+            return 1;
+        } else {
+            x = lua_tonumber(L, counter);
+            counter++;
+        }
+
+    if (!lua_isnumber(L, counter)) {
         lua_pushstring(L, "createLabel: wrong argument type");
         lua_error(L);
         return 1;
     } else {
-        y = lua_tonumber(L, 3);
+        y = lua_tonumber(L, counter);
+        counter++;
     }
-    if (!lua_isnumber(L, 4)) {
+
+    if (!lua_isnumber(L, counter)) {
         lua_pushstring(L, "createLabel: wrong argument type");
         lua_error(L);
         return 1;
     } else {
-        width = lua_tonumber(L, 4);
+        width = lua_tonumber(L, counter);
+        counter++;
     }
-    if (!lua_isnumber(L, 5)) {
+
+    if (!lua_isnumber(L, counter)) {
         lua_pushstring(L, "createLabel: wrong argument type");
         lua_error(L);
         return 1;
     } else {
-        height = lua_tonumber(L, 5);
+        height = lua_tonumber(L, counter);
+        counter++;
     }
-    if (!lua_isnumber(L, 6)) {
+
+    if (!lua_isnumber(L, counter)) {
         lua_pushstring(L, "createLabel: wrong argument type");
         lua_error(L);
         return 1;
     } else {
-        fillBackground = lua_toboolean(L, 6);
+        fillBackground = lua_toboolean(L, counter);
+        counter++;
     }
-    if (lua_gettop(L) > 6) {
-        if (!lua_isboolean(L, 7)) {
+    if (lua_gettop(L) > counter) {
+        counter++;
+        if (!lua_isboolean(L, counter)) {
             lua_pushstring(L, "createLabel: wrong argument type");
             lua_error(L);
             return 1;
         } else {
-            clickthrough = lua_toboolean(L, 7);
+            clickthrough = lua_toboolean(L, counter);
         }
     }
     Host& host = getHostFromLua(L);
     QString name(luaSendText.c_str());
-    lua_pushboolean(L, mudlet::self()->createLabel(&host, name, x, y, width, height, fillBackground, clickthrough));
+    QString windowname(luaSendWindow.c_str());
+    auto pC = host.mpConsole->mSubConsoleMap.value(name);
```

Comment:
Yes, I did that because creating a Label with the same name as a UserWindow causes the Userwindow to break down. Should be solved in another way and I have to write an Issue.